File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    max_w = 0  # 긴 변들 중 가장 큰 값 (지갑의 가로)
    max_h = 0  # 짧은 변들 중 가장 큰 값 (지갑의 세로)
    
    for i in range(len(sizes)):

        current_max = max(sizes[i][0], sizes[i][1])
        current_min = min(sizes[i][0], sizes[i][1])
        
        if current_max > max_w:
            max_w = current_max
            
        if current_min > max_h:
            max_h = current_min
            
    return max_w * max_h

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        
        int max_w = 0;
        int max_h = 0;
        
        for(int[] card : sizes){
            int w = Math.max(card[0], card[1]);
            int h = Math.min(card[0], card[1]);
            
            max_w = Math.max(max_w, w);
            max_h = Math.max(max_h, h);
        }
        
        int answer = 0;
        
        answer = max_w * max_h;
        
        return answer;
    }
}